### PR TITLE
Various Updates from Discord

### PR DESCRIPTION
**Pull request by gmason0**
This pull requests involves a few changes from users in the discord.

Bug fixes:
1. Multiple Plugins: Reset Antiban settings when shutting down plugins that support Antiban
2. AutoWoodcutting: Fix ActionCooldown not triggering correctly (oops)
3. Shooting Star Plugin: Change to AFKing at star location instead of AFKing at bank when star disappears

Features:
1. Rs2Bank: Adjust walkTo bank methods to allow to pass a BankLocation ( still kept other methods using nearestbank )

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java
Comment (See1Duck) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java:1]: Still missing the check for if the action cooldown is active, you need to add this:
```Java
if(Rs2AntibanSettings.actionCooldownActive)
                    return;
```

Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java:1]: Sorry I have missed this, this was added in commit [6c2737d](https://github.com/chsami/microbot/pull/383/commits/6c2737d91c533c63be85d1afb6b7a7bcf71d83c6)

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java [hunk 1]
```diff
@@ -1160,31 +1193,41 @@ public static boolean handleBankPin(String pin) {
...
      * @param emptySlotCount
      * @param distance
-     *
      * @return
      */
-    public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, WorldPoint initialPlayerLocation, int emptySlotCount, int distance) {
+    public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, BankLocation bankLocation, WorldPoint initialPlayerLocation, int emptySlotCount, int distance) {
         if (Rs2Inventory.getEmptySlots() <= emptySlotCount) {
-            boolean isBankOpen = Rs2Bank.walkToBankAndUseBank();
+            boolean isBankOpen = Rs2Bank.walkToBankAndUseBank(bankLocation);
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:1230]: why are we passing banklocation here? Shouldn't we by default always use the closest available bank?

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java [hunk 2]
```diff
@@ -1090,31 +1090,64 @@ public static BankLocation getNearestBank() {
...
+        return walkToBank(getNearestBank());
+    }
+
+    /**
+     * Walk to bank location
+     * 
+     * @param bankLocation 
+     * @return true if player location is less than 4 tiles away from the bank location
+     */
+    public static boolean walkToBank(BankLocation bankLocation) {
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:1102]: Why do we have to pass a banklocation? shouldn't we by default take the closest available bank?

**Review by chsami — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java (hunk 1 above)
Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:1230]: I think the Github change view may not be giving the entire picture here... The adjustments I made around this was to adjust the original methods to allow someone to pass a BankLocation in from BankLocation.java... 

IF you see, I had created more overloaded methods to maintain combability with previous scripts that by default, will go to NearestBank Location.

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java (hunk 2 above)
Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:1102]: See the above comment...

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java (hunk 1 above)
Comment (chsami) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java:1230]: Oh my bad, wasn't clear from the github change view. I understand now.

**Review by chsami — APPROVED** [review]

**Pull request merged**